gpu4pyscf/lib/cupy_helper.py

```diff
@@ -82,6 +83,51 @@ def get_avail_mem():
         mem_avail = cupy.cuda.runtime.memGetInfo()[0]
         return mem_avail + total_mem - used_mem
 
+def broadcast_to_devices():
+    ''' Broadcast cupy ndarray to all the devices, return a list of cupy ndarray
+    '''
+
+
+def reduce_to_device(array_list):
+    ''' Reduce a list of ndarray in different devices to device 0
+    TODO: reduce memory footprint, improve throughput
+    '''
+    num_devices = len(array_list)
```

Comment:
* early return for single device
* Assert len(array_list) == cudaGetDeviceCount